src/logplayer/player.cpp

```diff
@@ -113,30 +118,60 @@ bool Player::good()
     return packets.size() > 0;
 }
 
-void Player::sendMessage(const Frame* packet)
+void Player::sendMessage(const Frame *packet)
 {
     RoboCup2014Legacy::Wrapper::SSL_WrapperPacket legacyVisionPacket;
     SSL_Referee refereePacket;
 
-    if (packet->type == MESSAGE_BLANK) {
+    if (packet->type == MESSAGE_BLANK)
+    {
         // ignore
-    } else if (packet->type == MESSAGE_UNKNOWN) {
+    }
+    else if (packet->type == MESSAGE_UNKNOWN)
+    {
         // OK, let's try to figure this out by parsing the message
-        if (refereePacket.ParseFromArray(packet->data.data(), packet->data.size())) {
+        if (refereePacket.ParseFromArray(packet->data.data(), packet->data.size()))
+        {
             m_referee->writeData(packet->data);
-        } else if (legacyVisionPacket.ParseFromArray(packet->data.data(), packet->data.size())) {
+        }
+        else if (legacyVisionPacket.ParseFromArray(packet->data.data(), packet->data.size()))
+        {
             m_legacyVision->writeData(packet->data);
-        } else {
+        }
+        else
+        {
             std::cout << "Error unsupported or corrupt packet found in log file!" << std::endl;
         }
-    } else if (packet->type == MESSAGE_SSL_VISION_2010) {
+    }
+    else if (packet->type == MESSAGE_SSL_VISION_2010)
+    {
         m_legacyVision->writeData(packet->data);
-    } else if (packet->type == MESSAGE_SSL_REFBOX_2013) {
+    }
+    else if (packet->type == MESSAGE_SSL_REFBOX_2013)
+    {
         m_referee->writeData(packet->data);
-    } else if (packet->type == MESSAGE_SSL_VISION_2014) {
+    }
+    else if (packet->type == MESSAGE_SSL_VISION_2014)
+    {
         m_vision->writeData(packet->data);
-    } else {
-        std::cout << "Error unsupported message type found in log file!" << std::endl;
+    }
+    else if (packet->type == MESSAGE_SSL_VISION_TRACKER_2020)
+    {
+        if (receive_VT)
+        {
+        std::cout << "MESSAGE_SSL_VISION_TRACKER_2020 Received" << std::endl;    
+        receive_VT=false;
+        }
+    }
+else if (packet->type == MESSAGE_SSL_INDEX_2021)
+    {
+        std::cout << "MESSAGE_SSL_INDEX_2021 Received" << std::endl;
+    }
+
+    else
+    {
+        std::cout << packet->type <<std::endl;
+        std::cout << "Warning unsupported message type found in log file!" << std::endl;
```

Comment:
Better move those two lines into one. Like "... in the log file: " << packet->type <<stdl::endl.

And it would nicer to print it on stderr (std::cerr).